### DISCUSSION TITLE
How to dubug an error code: Error while trying to publish Module (aptos move publish): "Simulation failed with status: Move abort in 0x3::token: 0x60001"

### DISCUSSION THREAD

**Discussion by kshitijc1506** [aptos-labs/aptos-developer-discussions#103] [Questions] [answered]
Asked by `0xhepha` on Discord (#dev-discussion)

> Im getting this error while trying to publish a module to the testnet with aptos move publish
> {
>   "Error": "Simulation failed with status: Move abort in 0x3::token: 0x60001"
> }
> 
> There is no info online about this.. any repo or docs? 
> at least a --verbose or something?

**Comment by kshitijc1506** [comment]
Could you provide with me your module code? I think I'll need that to debug here. It looks like it is failing a check/assertion in the `0x3::token` module.

**Reply by gregnazario** [reply, marked as answer]
## How to debug an error code

In this case, let's go through the breakdown:

```
{
"Error": "Simulation failed with status: Move abort in 0x3::token: 0x60001"
}
```

1. `Simulation failed with status` -> This means simulation failed, and no transaction was submitted to fail.  Means that if the transaction had been actually submitted, it would have failed (but there's no gas spent)
2. `Move abort in 0x3::token` -> We have an abort in `0x3::token`. In this case, you can find this at the `0x3` address, under module `token`.  You can see this source in the explorer https://explorer.aptoslabs.com/account/0x3/modules/code/token?network=mainnet
3. `0x60001` This means it's error code `0x60001`.  This can be one of two things, someone has actually made a really large error code, or there are error typings.

In this case `0x60000 = not_found` https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/move-stdlib/sources/error.move#L39-L40.

Now, that we've determined it's `not_found`, the other part is `1`.  We simply have to look in the `token` module for an error code `1`.

We find that it's the code `ECOLLECTIONS_NOT_PUBLISHED`, which means that the address being passed for the collection, has never published a collection. https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move#L48-L49

## But, why is it not telling me `ECOLLECTIONS_NOT_PUBLISHED` and the proper error message?

This message is coming from the Aptos CLI.  I suspect your version may be out of date, as newer versions should handle the error codes correctly.

To check your Aptos CLI version, you can run `aptos info`, which will tell you full build information.

### Update CLI

For Windows / Linux you can run `aptos update` see: https://aptos.dev/tools/aptos-cli/install-cli/automated-install#update

For Mac you can run `brew update; brew upgrade aptos` https://aptos.dev/tools/aptos-cli/install-cli/install-from-brew
